processing/src/main/java/io/druid/query/search/search/DefaultSearchQueryMetrics.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.search.search;
+
+import com.metamx.emitter.service.ServiceEmitter;
+import io.druid.collections.bitmap.BitmapFactory;
+import io.druid.query.BitmapResultFactory;
+import io.druid.query.Query;
+import io.druid.query.QueryMetrics;
+import io.druid.query.filter.Filter;
+import org.joda.time.Interval;
+
+import java.util.List;
+
+public class DefaultSearchQueryMetrics implements SearchQueryMetrics
+{
+  private QueryMetrics<Query<?>> queryMetrics;
+
+  public DefaultSearchQueryMetrics(QueryMetrics<Query<?>> queryMetrics)
+  {
+    this.queryMetrics = queryMetrics;
+  }
+
+  @Override
+  public void query(SearchQuery query)
+  {
+    dataSource(query);
```

Comment:
I think the body of this `query()` method should be empty, because `query()` should be already called on the provided delegate `QueryMetrics`, according to the `GenericQueryMetricsFactory` documentation. Please reflect this in a doc comment to the `DefaultSearchQueryMetrics()` constructor, like `queryMetrics.query(query) must already be called on the provided queryMetrics`